middleware.js

```diff
@@ -92,9 +93,14 @@ module.exports = function(options) {
 
   var maxAge = options.maxAge || 0;
 
+    
   //Allow custom log function or default one
-  var log = options.log || function (key, val) {
-    console.error('  \x1B[90m%s:\x1B[0m \x1B[36m%s\x1B[0m', key, val);
+  var log = options.log || function (severity, key, val) {
+    if (typeof(console[severity]) === 'function') {
+        console[severity]('  \x1B[90m%s:\x1B[0m \x1B[36m%s\x1B[0m', key, val);
+    } else {
+        console.error('  \x1B[90m%s:\x1B[0m \x1B[36m%s\x1B[0m', key, val);
```

Comment:
`debug` makes sense. Thanks for updating the call sites with more logical choice. 👍 

> Also having this check is more secure if in further developments, new call to log are added

Perhaps we should *enforce* that it should be a valid method name, let the CI build break and hence block the pull request. Generally, carrying dead code paths in code base is something discouraged by code coverage conscious parties. 😃 

(I intend to add cli test mechanics after merging this pull request)